x/incentives/keeper/distribute_test.go

```diff
@@ -868,3 +871,199 @@ func (s *KeeperTestSuite) TestGetPoolFromGaugeId() {
 		})
 	}
 }
+
+// TestFunctionalInternalExternalCLGauge is a functional test that covers more complex scenarios relating to distributing incentives through gauges
+// at the end of each epoch.
+//
+//
+// Testing strategy:
+// 1. Initialize variables.
+// 2. Setup CL pool and gauge (gauge automatically gets created at the end of CL pool creation).
+// 3. Create external no-lock gauges for CL pools
+// 4. Create Distribution records to incentivize internal CL no-lock gauges
```

Comment:
We currently seem to internally incentivize both pools but should we only be incentivizing one?

Reply:
so the whole test adds internal incentives to both pools. Is there a specific reason why you only wanted one pool to be internally incentivized?

Reply:
I would like to test that when a pool does not have internal incentives enabled, nothing breaks.

Maybe we could instead add a third pool and validate it instead of messing with the existing logic?

Reply:
added ClPool3
- This pool has no internal incentives 
- This pool has external incentive distributed over 1 epoch
- did validation checks on it 
https://github.com/osmosis-labs/osmosis/pull/5500/commits/aab545b61b67e73c6039be2d8c2ec11da95080b0